custom_components/waste_collection_schedule/waste_collection_schedule/source/gastrikeatervinnare_se.py

```diff
@@ -0,0 +1,69 @@
+import requests
+from bs4 import BeautifulSoup
+from waste_collection_schedule import Collection
+from datetime import date
+
+TITLE = "Gästrike Återvinnare"
+DESCRIPTION = "Source for Gästrike Återvinnare waste collection"
+URL = "https://gastrikeatervinnare.se/"
+API_URL = (
+    "https://gastrikeatervinnare.se/wp-admin/admin-ajax.php"
+)
+TEST_CASES = {
+    "Groceries Årsunda": {"street": "Nedre Vägen 52", "city": "Årsunda"},
+    "Police Sandviken": {"street": "Bryggargatan 6", "city": "Sandviken"},
+    "Police Gävle": {"street": "Södra Centralgatan 1", "city": "Gävle"},
+    "Library Ockelbo": {"street": "Södra Åsgatan 30D", "city": "Ockelbo"},
+}
+ICON_MAP = {"Restavfall": "mdi:trash-can", "Matavfall": "mdi:leaf", "Blandat": "mdi:trash-can"}
+
+class Source:
+    def __init__(self, street, city):
+        self._street = street.strip().lower()
+        self._city = city.strip().lower()
+
+    def fetch(self):
+        data = {"action": "pickup_search", "query": self._street}
+        response = requests.post(API_URL, data=data)
+        response.encoding = "UTF-8"
+        soup = BeautifulSoup(response.text, "html.parser")
+        addresses = soup.find_all(
+            "p", attrs={"class": "pickup-adress"}
+        )
+        infos = soup.find_all("div", attrs={"class": "pickup-types"})
+
+        for (addressTag, info) in zip(addresses, infos):
+            street = addressTag.text.split(",")[0].strip().lower()
+            city = addressTag.find("span", attrs={"class": "pickup-locality"}).text.strip().lower()
+            if street == self._street and city == self._city:
+                return self.get_entries(info)
+
+        return []
```

Comment:
As this indicates a miss configured source (wrong street or city) I would prefer to raise an Exception as this will be shown in the logs but just returning no entries is hard to debug/understand what's wrong.

Reply:
Agree, I can look into that. The current code is mainly just copied from an existing source (tekniskaverken_se) and modified to fit the html response from Gästrike Återvinnare. 

But I can look at some other sources and see how they do it